roles/openshift_facts/library/openshift_facts.py

```diff
@@ -1402,11 +1402,14 @@ def set_proxy_facts(facts):
             builddefaults['git_http_proxy'] = builddefaults['http_proxy']
         if 'git_https_proxy' not in builddefaults and 'https_proxy' in builddefaults:
             builddefaults['git_https_proxy'] = builddefaults['https_proxy']
-        if 'admission_plugin_config' not in builddefaults:
-            builddefaults['admission_plugin_config'] = dict()
+        # If we're actually defining a proxy config then create kube_admission_plugin_config
+        # if it doesn't exist, then merge builddefaults[config] structure
+        # into kube_admission_plugin_config
+        if 'kube_admission_plugin_config' not in facts['master']:
+            facts['master']['kube_admission_plugin_config'] = dict()    
         if 'config' in builddefaults and ('http_proxy' in builddefaults or \
                 'https_proxy' in builddefaults):
-            facts['master']['admission_plugin_config'].update(builddefaults['config'])
+            facts['master']['kube_admission_plugin_config'].update(builddefaults['config'])
```

Comment:
Oh, that's the sort of messy thing. So builddefaults.config yaml structure is defined here 
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_builddefaults/vars/main.yml
and it depends on other builddefaults values being set, so we run the builddefaults openshift_facts role twice here
https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_builddefaults/tasks/main.yml

I guess I could move the yaml structure into openshift_facts.py but I really dislike having that embedded this deeply. What do you think? Or maybe I'm misunderstanding what you're suggesting.